src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -90,19 +90,50 @@ def upgrade_status(self) -> orchestrator.UpgradeStatusSpec:
                 r.message = 'Upgrade paused'
         return r
 
+    def _check_target_version(self, version: str) -> Optional[str]:
+        try:
+            (major, minor, patch) = version.split('.')
+            assert int(minor) >= 0
+            assert int(patch) >= 0
+        except:
+            return 'version must be in the form X.Y.Z (e.g., 15.2.3)'
+        if int(major) < 15 or (int(major) == 15 and int(minor) < 2):
+            return 'cephadm only supports octopus (15.2.0) or later'
+
+        # to far a jump?
+        current_version = self.mgr.version.split('ceph version ')[1]
+        current_major, current_minor, current_patch = current_version.split('-')[0].split('.')
+        if int(current_major) < int(major) - 2:
+            return f'ceph can only upgrade 1 or 2 major versions at a time; {current_version} -> {version} is too big a jump'
+        if int(current_major) > int(major):
+            return f'ceph cannot downgrade major versions (from {current_version} to {version})'
+        if int(current_major) == int(major):
+            if int(current_minor) > int(minor):
+                return f'ceph cannot downgrade to a {"rc" if minor == "1" else "dev"} release'
```

Comment:
:-1: for special handing of rc and dev releases. 

```suggestion
                return f'ceph cannot downgrade versions (from {current_version} to {version})'
```

Reply:
I want to allow downgrade of point releases, though...